File: frontend/src/services/playlist.js
import api from "./api";

export async function getPlaylists(userId) {
    const res = await api.get(`/playlist/${userId}`);
    return res.data;
}

export async function createPlaylist(name, userId) {
    const res = await api.post("/playlist/create", {
        name,
        user_id: userId,
    });

    return res.data;
}

export async function deletePlaylist(id) {
    const res = await api.delete(`/playlist/${id}`);
    return res.data;
}

File: backend/app/api/playlist.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.models.playlist import Playlist
from app.schemas.playlist import PlaylistCreate

router = APIRouter(
    prefix="/playlist",
    tags=["Playlist"]
)


@router.post("/create")
def create_playlist(
    playlist: PlaylistCreate,
    db: Session = Depends(get_db)
):

    new_playlist = Playlist(
        name=playlist.name,
        user_id=playlist.user_id
    )

    db.add(new_playlist)
    db.commit()
    db.refresh(new_playlist)

    return new_playlist


@router.get("/{user_id}")
def get_playlists(
    user_id: int,
    db: Session = Depends(get_db)
):

    playlists = db.query(Playlist).filter(
        Playlist.user_id == user_id
    ).all()

    return playlists


@router.delete("/{playlist_id}")
def delete_playlist(
    playlist_id: int,
    db: Session = Depends(get_db)
):

    playlist = db.query(Playlist).filter(
        Playlist.id == playlist_id
    ).first()

    if not playlist:
        raise HTTPException(
            status_code=404,
            detail="Playlist not found"
        )

    db.delete(playlist)
    db.commit()

    return {"message": "Playlist deleted"}